### PR TITLE
Got rid of all xfails since the tests should pass now

### PR DESCRIPTION
**Pull request by jessberg**

### CODE REVIEW

File: test_env.py [hunk 1]
```diff
@@ -95,3 +95,4 @@ class TestClassSimulator:
     def test_deployment(self, query_file, query_udfs):
         result = check_filter(query_file, query_udfs)
         assert result == util.EXIT_SUCCESS
+        #TODO: check the results in result.txt as well
```
Comment (fruffy) [inline test_env.py:98]: Ideally, we would write dedicate rust or python tests instead of relying on results.txt. We should have an API to query results.

**Review by fruffy — APPROVED** [review]

File: test_env.py (hunk 1 above)
Comment (jessberg) [reply test_env.py:98]: Do you mean use another file that is used inside of test_env, like check_filter?  Also, as far as an API goes, how would that work, given that the simulator process will be dead by the time we want to look at results?

Comment (fruffy) [reply test_env.py:98]: Write tests that use the simulator API directly. 
```bash
    for tick in 0..20 {
        simulator.tick(tick);
    }
```
In this loop we would query results from storage and assert if they are correct.

Comment (jessberg) [reply test_env.py:98]: Okay, thanks for the info.

**Pull request merged**